modules/standard/Math.chpl

```diff
@@ -1057,6 +1057,66 @@ module Math {
     return truncf(x);
   }
 
+
+  /* Returns the Bessel function of the first kind of order `0` of `x`. */
+  extern proc j0(x: real(64)): real(64);
+
+  /* Returns the Bessel function of the first kind of order `0` of `x`. */
+  inline proc j0(x : real(32)): real(32) {
+    extern proc j0f(x: real(32)): real(32);
+    return j0f(x);
+  }
+
+
+  /* Returns the Bessel function of the first kind of order `1` of `x`. */
+  extern proc j1(x: real(64)): real(64);
+
+  /* Returns the Bessel function of the first kind of order `1` of `x`. */
+  inline proc j1(x : real(32)): real(32) {
+    extern proc j1f(x: real(32)): real(32);
+    return j1f(x);
+  }
+
+
+  /* Returns the Bessel function of the first kind of order `n` of `x`. */
+  extern proc jn(n: c_int, x: real(64)): real(64);
+
+  /* Returns the Bessel function of the first kind of order `n` of `x`. */
+  inline proc jn(n: c_int, x : real(32)): real(32) {
+    extern proc jnf(n: c_int, x: real(32)): real(32);
+    return jnf(n, x);
+  }
+
+
+  /* Returns the Bessel function of the second kind of order `0` of `x`. */
+  extern proc y0(x: real(64)): real(64);
+
+  /* Returns the Bessel function of the second kind of order `0` of `x`. */
+  inline proc y0(x : real(32)): real(32) {
+    extern proc y0f(x: real(32)): real(32);
+    return y0f(x);
+  }
+
+
+  /* Returns the Bessel function of the second kind of order `1` of `x`. */
```

Comment:
Let's document that negative values of `x` are not supported.

Reply:
yes, its good, then I think a test for only positive values will be cool for bessel function of second kind, what should be done for negetive values, an exception is thrown or what?

Reply:
We might want this to throw an error eventually (@psahabu can confirm if that's our intention in `Math.chpl` eventually), but for now, let's stay consistent with our procedures in the module by halting if `x < 0`.

Reply:
yes, I did